src/test/java/com/spotify/docker/client/DefaultDockerClientUnitTest.java

```diff
@@ -137,62 +106,40 @@ public void testHostForIpHttps() {
     assertThat(client.getHost(), equalTo("192.168.53.103"));
   }
 
-  @Test
-  public void testNoHeaders() throws Exception {
-    final DefaultDockerClient dockerClient = new DefaultDockerClient(
-        builder, clientBuilderSupplier);
-    dockerClient.info();
-
-    verify(builderMock, never()).header(anyString(), anyString());
+  private RecordedRequest takeRequestImmediately() throws InterruptedException {
+    return server.takeRequest(1, TimeUnit.MILLISECONDS);
   }
 
   @Test
-  public void testOneHeader() throws Exception {
-    builder.header("foo", 1);
+  public void testCustomHeaders() throws Exception {
+    builder.header("int", 1);
+    builder.header("string", "2");
+    builder.header("list", Lists.newArrayList("a", "b", "c"));
 
-    final DefaultDockerClient dockerClient = new DefaultDockerClient(
-        builder, clientBuilderSupplier);
+    server.enqueue(new MockResponse());
+
+    final DefaultDockerClient dockerClient = new DefaultDockerClient(builder);
     dockerClient.info();
 
-    final ArgumentCaptor<String> keyArgument = ArgumentCaptor.forClass(String.class);
-    final ArgumentCaptor<Object> valueArgument = ArgumentCaptor.forClass(Object.class);
-    verify(builderMock, times(1)).header(keyArgument.capture(), valueArgument.capture());
+    final RecordedRequest recordedRequest = takeRequestImmediately();
+    assertThat(recordedRequest.getMethod(), is("GET"));
+    assertThat(recordedRequest.getPath(), is("/info"));
 
-    Assert.assertEquals("foo", keyArgument.getValue());
-    Assert.assertEquals(1, valueArgument.getValue());
+    assertThat(recordedRequest.getHeader("int"), is("1"));
+    assertThat(recordedRequest.getHeader("string"), is("2"));
+    // TODO (mbrown): this seems like incorrect behavior - the client should send 3 headers with
```

Comment:
it is an existing bug in this Builder's "Add header" method which I didn't want to fix as a part of this, but I think should be fixed at some point